subsys/net/lib/openthread/platform/radio.c

```diff
@@ -298,7 +298,13 @@ void transmit_message(struct k_work *tx_job)
 	channel = sTransmitFrame.mChannel;
 
 	radio_api->set_channel(radio_dev, sTransmitFrame.mChannel);
+
+#if defined(CONFIG_IEEE802154_SELECTIVE_TXPOWER)
+	net_pkt_set_ieee802154_txpwr(tx_pkt, get_transmit_power_for_channel(channel));
+	net_pkt_set_ieee802154_txpwr_use(tx_pkt, true);
```

Comment:
@edmont raised the same issue before. The reason I wanted to avoid introducing such a macro was that it cannot be easily passed between generic interfaces connected through pointers to APIs, so very similar macros would have to be created in other parts of the system. However, `struct net_pkt` is used by higher networking layers as well. If the decision about TX power is made when `struct net_pkt` is already available, then we could introduce a macro in `net_pkt.h` such as `NET_PKT_IEEE802154_TXPWR_IGNORE` and use it instead of the flag. I agree that it would be cleaner (no redundancy). @edmont @ahasztag what are your opinions on this?

Reply:
Wait, my question is different. As I understand when we call `net_pkt_set_ieee802154_txpwr` then we _always_ do it with the intention to use that power - otherwise there is no point of calling this function. No magic value needed.

Reply:
Note this is different than the case in the radio driver when the value is passed by metadata  there was no separate function for setting the txpwr for a given packet

Reply:
> Wait, my question is different.

Sorry @ahasztag , I haven't read your comment carefully enough. I get that the redundancy is irritating, but I'm wondering if "optimizing" it will not cause more chaos than this additional line of code. Suppose you're preparing a packet and you want to ensure that the default TX power will be used to transmit it. If we remove the setter of `ieee802154_txpwr_use` bit, the only option is to rely on the field to be _already_ set to zero e.g. on initialization. That may not be true, for instance in case of reusing some static `struct net_pkt`.